lib/queue.rb

```diff
@@ -1,31 +1,49 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
-  end
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if @front == @back
+      #decide
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
 
-  def dequeue
-    raise NotImplementedError, "Not yet implemented"
   end
+  
+  def dequeue
+    index = @front
+    element = @store[index]
+    
+    @store[index] = nil
 
+    @front = (@front + 1) % @store.length 
+    return element
+  end
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.compact.length
   end
```

Comment:
This works, but it's quite expensive O(n) for both space and time and you lose some of the benefits of the circular buffer.